File: code/generate_strategies/RA(method)/python/function_bokeh_circle.py
from bokeh.plotting import figure, show, output_file

class Plotter:
    def create_circle_plot(self, x_values: str, y_values: str, radius: int = 10, title: str = "Circle Plot") -> str:
        
        if not x_values or not y_values:
            return "Error: x_values and y_values must not be empty."
        
        try:
            x_values_list = x_values.strip('[]').split(',')
            y_values_list = y_values.strip('[]').split(',')
            x_values_list = [x.strip().strip('"') for x in x_values_list]
            y_values_list = [int(y.strip()) for y in y_values_list]
        except ValueError:
            return "Error: Invalid input format."

        if len(x_values_list) != len(y_values_list):
            return "Error: x_values and y_values must have the same length."
        
        p = figure(title=title, x_axis_label='X Axis', y_axis_label='Y Axis')
        p.circle(x_values_list, y_values_list, size=radius, color="green", alpha=0.5)
        
        output_file("circle_plot.html")
        show(p)
        
        return "Plot displayed."

File: code/generate_strategies/RA(method)/java/function_bokeh_circle.java
package com.example;

import org.jfree.chart.ChartFactory;
import org.jfree.chart.ChartUtils;
import org.jfree.chart.JFreeChart;
import org.jfree.data.xy.XYSeries;
import org.jfree.data.xy.XYSeriesCollection;
import java.io.File;
import java.io.IOException;

public class Plotter {
    public String create_circle_plot(String x_values, String y_values, int radius, String title) {
        class Plotter {
            public String createPlot(String x_values, String y_values, int radius, String title) {
                if (x_values == null || y_values == null || x_values.isEmpty() || y_values.isEmpty()) {
                    return "Error: x_values and y_values must not be empty.";
                }

                try {
                    String[] x_values_list = x_values.replace("[", "").replace("]", "").split(",");
                    String[] y_values_list = y_values.replace("[", "").replace("]", "").split(",");

                    if (x_values_list.length != y_values_list.length) {
                        return "Error: x_values and y_values must have the same length.";
                    }

                    XYSeries series = new XYSeries("Data");
                    for (int i = 0; i < x_values_list.length; i++) {
                        String x = x_values_list[i].trim().replace("\"", "");
                        int y = Integer.parseInt(y_values_list[i].trim());
                        series.add(i, y); // Using index as x since we have string x values
                    }

                    XYSeriesCollection dataset = new XYSeriesCollection();
                    dataset.addSeries(series);

                    JFreeChart chart = ChartFactory.createScatterPlot(
                            title,
                            "X Axis",
                            "Y Axis",
                            dataset
                    );

                    try {
                        ChartUtils.saveChartAsJPEG(new File("circle_plot.jpg"), chart, 800, 600);
                        return "Plot displayed.";
                    } catch (IOException e) {
                        return "Error: Could not save plot.";
                    }
                } catch (NumberFormatException e) {
                    return "Error: Invalid input format.";
                }
            }
        }

        return new Plotter().createPlot(x_values, y_values, radius != 0 ? radius : 10,
                title != null && !title.isEmpty() ? title : "Circle Plot");
    }
}